samcli/lib/providers/api_provider.py

```diff
@@ -25,20 +25,17 @@ def __init__(self, template_dict, parameter_overrides=None, cwd=None):
 
         Parameters
         ----------
-        template_dict : dict
-            Template as a dictionary
+        stacks : dict
+            List of stacks apis are extracted from
 
         cwd : str
             Optional working directory with respect to which we will resolve relative path to Swagger file
         """
-        self.template_dict = SamBaseProvider.get_template(template_dict, parameter_overrides)
-        self.resources = self.template_dict.get("Resources", {})
-
-        LOG.debug("%d resources found in the template", len(self.resources))
+        self.stacks = stacks
```

Comment:
Did we miss the parameters value substitution in `stack.resources` https://github.com/aws/aws-sam-cli/blob/12efe15b682926b006ce2a846d7cc6c71f5f2d74/samcli/lib/providers/sam_base_provider.py#L162

Reply:
I did not forgot:

```
@property
def resources(self) -> Dict:
    """
    Return the resources dictionary where SAM plugins have been run
    and parameter values have been substituted.
    """
    processed_template_dict: Dict = SamBaseProvider.get_template(self.template_dict, self.parameters)
    resources: Dict = processed_template_dict.get("Resources", {})
    return resources
```

There are a few integ tests to verify it like `test_invoke_with_env_using_parameters_with_custom_region()`